gui/choices.py

```diff
@@ -1163,20 +1163,22 @@ def __iter__(self):
 
 class CIFS_VFS_OBJECTS(object):
     def __init__(self):
-        self.__vfs_module_path = '/usr/local/lib/shared-modules/vfs'
+        self.__vfs_modules_path = '/usr/local/lib/shared-modules/vfs'
         self.__vfs_modules = []
-        self.__vfs_exclude = [
-            'shadow_copy2',
-            'recycle',
-        ]
-
-        if os.path.exists(self.__vfs_module_path):
-            for f in os.listdir(self.__vfs_module_path):
-                f = f.replace('.so', '')
-                if f not in self.__vfs_exclude:
-                    self.__vfs_modules.append(f)
+        self.__vfs_exclude = {'shadow_copy2', 'recycle', 'aio_pthread'}
+
+        if os.path.exists(self.__vfs_modules_path):
+            self.__vfs_modules.extend(
+                filter(
+                    lambda m: m not in self.__vfs_exclude,
+                    map(
+                        lambda f: f.rpartition('.')[0],
+                        os.listdir(self.__vfs_modules_path)
+                    )
+                )
+            )
         else:
-            self.__vfs_modules += ['streams_xattr', 'aio_pthread']
+            self.__vfs_modules.extend('streams_xattr')
```

Comment:
You cannt `extend` with a single item. I believe you're looking for .append here.